library/core/src/main/java/com/google/android/exoplayer2/video/MediaCodecVideoRenderer.java

```diff
@@ -1750,7 +1753,8 @@ public void onFrameRendered(MediaCodec codec, long presentationTimeUs, long nano
         // Stale event.
         return;
       }
```

Comment:
This stall event needs to be done in the lambda too.

Reply:
You have marked this done, have you have not change that code. Have you forgotten to upload a patch?

Reply:
@krocard I've put the check for listener changed (stale event), but also cleaned up the handler to avoid leaks and prevent the stale event from ever happening...  If you agree we can remove the stale check, but kind of hard to ever test.